docs/blueprints/promo-codes.md

```diff
@@ -0,0 +1,117 @@
+# Promo Codes 
+
+## Abstract
+
+Enable users to apply a promotional code to their account so that they can automatically receive the free credit associated with that code.
+
+Summary: When Sales/Marketing whats to run a promotion or hand out a specific amount of free credit to a group of people, we want to be able to generate a promo code that a user could enter into their account, so that specific free credits and expiration of the credits would be automatically applied to that user’s account.
+
+## Background
+
+
+## Design
+
+There are 2 high level aspects of this design:
+
+### Generating Promo Codes
+
+The satellite admin can generate promo codes from the admin UI. Once generated, these promo codes can be applied to any user's account.
+
+### Applying a Promo Code to a User
+
+On account creation or from the payments panel, a user should be able to insert a promo code and immediately have their project usage limits increased. If the promo code has an expiration associated with it, these limit increases and this promo code will disappear after the expiration date.
+
+## Rationale
+
+In the design detailed below, two additional database tables are defined. In theory, we could just have the `promo_codes` table and add some additional fields to the `users` table to get rid of the `user_promo_codes` table requirement. However, this will put more load on the `users` table, and deleting promo codes from users will have worse performance.
+
+## Implementation
+
+### Database Layer
+
+We will need a new table for promo codes, called `promo_codes`. The table will have the following fields:
+```
+name        string (unique)
```

Comment:
I would highly vote to make this a UUID and have the name be a secondary, but required unique field.

Reply:
done